src/GithubFilesBundle/Service/GithubImport.php

```diff
@@ -81,53 +81,46 @@ public function parseGithubDocument($lsDocKeys, $lsItemKeys, $fileContent){
+    public function saveCSVGithubDocument($lsDocKeys, $lsItemKeys, $content){
         $em = $this->getEntityManager();
         $lsDoc = new LsDoc();
 
-        foreach($content as $i => $row){
-            if($i < 1){
-                //create the new LsDoc
-                $lsDocIdentifier = Uuid::uuid5(Uuid::NAMESPACE_URL, $this->getValue($lsDocKeys['title'], $row))->toString();
-                $lsDoc->setIdentifier($lsDocIdentifier);
-                $lsDoc->setUri('local:'.$lsDocIdentifier);
-
-                if(empty($this->getValue($lsDocKeys['creator'], $row))){
-                    $lsDoc->setCreator('Imported from GitHub');
-                }else{
-                    $lsDoc->setCreator($this->getValue($lsDocKeys['creator'], $row));
-                }
-
-                $lsDoc->setTitle($this->getValue($lsDocKeys['title'], $row));
-
-                $lsDoc->setOfficialUri($this->getValue($lsDocKeys['officialSourceURL'], $row));
-                $lsDoc->setPublisher($this->getValue($lsDocKeys['publisher'], $row));
-                $lsDoc->setDescription($this->getValue($lsDocKeys['description'], $row));
-                $lsDoc->setVersion($this->getValue($lsDocKeys['version'], $row));
-                $lsDoc->setSubject($this->getValue($lsDocKeys['subject'], $row));
-                $lsDoc->setLanguage($this->getValue($lsDocKeys['language'], $row));
-                $lsDoc->setNote($this->getValue($lsDocKeys['notes'], $row));
-
-                $em->persist($lsDoc);
-                $em->flush();
-                continue;
-            }
+        //create the new LsDoc
+        $lsDoc->setUri('local:'.$lsDoc->getIdentifier());
 
-            $lsItem = $this->parseGithubStandard($lsDoc, $lsItemKeys, $row);
-            //create a new association between the item and the doc
+        if(empty($this->getValue($lsDocKeys['creator'], $content[0]))){
+            $lsDoc->setCreator('Imported from GitHub');
+        }else{
+            $lsDoc->setCreator($this->getValue($lsDocKeys['creator'], $content[0]));
         }
+
+        $lsDoc->setTitle($this->getValue($lsDocKeys['title'], $content[0]));
+        $lsDoc->setOfficialUri($this->getValue($lsDocKeys['officialSourceURL'], $content[0]));
+        $lsDoc->setPublisher($this->getValue($lsDocKeys['publisher'], $content[0]));
+        $lsDoc->setDescription($this->getValue($lsDocKeys['description'], $content[0]));
+        $lsDoc->setVersion($this->getValue($lsDocKeys['version'], $content[0]));
+        $lsDoc->setSubject($this->getValue($lsDocKeys['subject'], $content[0]));
+        $lsDoc->setLanguage($this->getValue($lsDocKeys['language'], $content[0]));
+        $lsDoc->setNote($this->getValue($lsDocKeys['notes'], $content[0]));
+
+        $em->persist($lsDoc);
+
+        for($i=1; $i < sizeof($content); $i++){
+            $row = $content[$i];
+            $lsItem = $this->parseCSVGithubStandard($lsDoc, $lsItemKeys, $row);
```

Comment:
`$lsItem` is never used, so we can probably not do the assignment here....

But, we may want to add them to an array if we look at doing another pass to add any parent/child associations from the data.
Or we could just do an explicit `$lsDoc->addTopLsItem($lsItem);` to have the document be the parent of all the statements (in general the items should be part of a parent/child hierarchy).